posydon/popsyn/synthetic_population.py

```diff
@@ -1,1094 +1,2660 @@
+                tmp.extend(group["event"].iloc[1:])
+                combined_events = "_".join(tmp)
+            else:
+                combined_events = "_".join(group["event"])
+            return pd.Series({"channel_debug": combined_events})
+
+        def mt_history(row):
+            if (
+                pd.notna(row["mt_history_HMS_HMS"])
+                and row["mt_history_HMS_HMS"] == "Stable contact phase"
+            ):
+                return row["channel"].replace("oRLO1", "oRLO1-contact")
+            # reverse is already included in the interpolation class now!
+            # elif (
+            #     pd.notna(row["mt_history_HMS_HMS"])
+            #     and row["mt_history_HMS_HMS"] == "Stable reverse mass-transfer phase"
+            # ):
+            #     return row["channel"].replace("oRLO1", "oRLO1-reverse")
+            else:
+                return row["channel"]
+
+        previous = 0
+
+        for i in tqdm(
+            range(0, len(unique_binary_indices), self.chunksize),
+            disable=not self.verbose,
+        ):
+            selection = unique_binary_indices[i : i + self.chunksize]
+
+            # create the dataframe for the chunk
+            df = pd.DataFrame(index=selection, columns=["channel_debug", "channel"])
+            end = (
+                previous
+                + self.history_lengths.iloc[i : i + self.chunksize].sum().iloc[0]
+            )
+
+            # get the history of chunk events and transform the interp_class_HMS_HMS
+            interp_class_HMS_HMS = self.oneline.select(
+                start=i, stop=i + self.chunksize, columns=["interp_class_HMS_HMS"]
+            )
+            events = self.history.select(start=previous, stop=end, columns=["event"])
+
+            mask = ~pd.isna(interp_class_HMS_HMS["interp_class_HMS_HMS"].values)
+            interp_class_HMS_HMS.loc[mask, "interp_class_HMS_HMS"] = (
+                interp_class_HMS_HMS[mask]
+                .apply(lambda x: HMS_HMS_event_dict[x["interp_class_HMS_HMS"]], axis=1)
+                .values
+            )
+            del mask
+
+            previous = end
+            # combine based on the index, this allows for an easier apply later
+            merged = pd.merge(
+                events.dropna(), interp_class_HMS_HMS, left_index=True, right_index=True
+            )
+            del events, interp_class_HMS_HMS
+
+            merged.index.name = "binary_index"
+            df["channel_debug"] = merged.groupby("binary_index").apply(get_events)
+            del merged
+            df["channel"] = (
+                df["channel_debug"]
+                .str.replace("_redirect_from_ZAMS", "")
+                .str.replace("_redirect_from_CO_HMS_RLO", "")
+                .str.replace("_redirect_from_CO_HeMS_RLO", "")
+                .str.replace("_redirect_from_CO_HeMS", "")
+                .str.replace("_CO_contact", "")
+            )
+
+            if mt_history:
```

Comment:
`mt_history` is a bool at the beginning of the function `calculate_formation_channels` but then it is a name of a function defined on line 1320. when `if mt_history` is called here what will happen? Could it be passes as just not False?

Reply:
Good catch! It probably means that it will **always** go into the `if` statement. I will change the function name.